server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueAssigner.java

```diff
@@ -59,24 +59,20 @@ public IssueAssigner(ScmInfoRepository scmInfoRepository, ScmAccountToUser scmAc
 
   @Override
   public void onIssue(Component component, DefaultIssue issue) {
-    if (issue.isNew()) {
```

Comment:
This is not correct. The algorithm must be:
1. if author is null
   1.1. load scm
   1.2. guess author
2. if assignee is null
    2.1. if author just set during step 1.2, guess assignee. Else that means that issue was manually unassigned by end-users and "unassigned" status must not be changed